### PR TITLE
[Ruby - In Review] Adding kind property on the resource model in ms_rest_azure. 

### PR DESCRIPTION
**Pull request by vishrutshah**
This is to resolve issue https://github.com/Azure/azure-sdk-ruby/issues/18.

### CODE REVIEW

**Comment by azurecla** [comment]
Hi __@vishrutshah__, I'm your friendly neighborhood Azure Pull Request Bot (You can call me AZPRBOT). Thanks for your contribution!
    <span>You've already signed the contribution license agreement. Thanks!</span>
        <span>We will now validate the agreement and then real humans will evaluate your PR.</span>

TTYL, AZPRBOT;

File: src/client/Ruby/ms-rest-azure/lib/ms_rest_azure/resource.rb [hunk 1]
```diff
@@ -23,6 +23,10 @@ class Resource
     # @return [Hash{String => String}] the tags attached to resources (optional).
     attr_accessor :tags
 
+    # @return [String] the kind of the resource.
+    # FIXME kind attribute was introduced because of SiteInstance model of azure_mgmt_web gem has kind attribute.
```
Comment (veronicagg) [inline src/client/Ruby/ms-rest-azure/lib/ms_rest_azure/resource.rb]: If 'kind' becomes part of resource, we may not need a 'fixme' comment.

**Comment by veronicagg** [comment]
@vishrutshah left one comment, change looks good to me, thanks!

File: src/client/Ruby/ms-rest-azure/lib/ms_rest_azure/resource.rb (hunk 1 above)
Comment (vishrutshah) [inline src/client/Ruby/ms-rest-azure/lib/ms_rest_azure/resource.rb]: @veronicagg I completely agree but i haven't found any swagger other than web doing this, so i was thinking to keep it until 'kind' officially becomes part of all resource.

Comment (devigned) [inline src/client/Ruby/ms-rest-azure/lib/ms_rest_azure/resource.rb]: Kind is in this [storage spec](https://github.com/Azure/azure-rest-api-specs/blob/master/arm-storage/2016-01-01/swagger/storage.json). Thoughts?

Comment (vishrutshah) [inline src/client/Ruby/ms-rest-azure/lib/ms_rest_azure/resource.rb]: @devigned I see the "kind" but it is not part of the Resource definition. azure_arm_web is adding "kind" in the [resource ](https://github.com/Azure/azure-rest-api-specs/blob/master/arm-web/2015-08-01/swagger/service.json#L21441)itself. Isn't that different?

**Comment by vishrutshah** [comment]
@veronicagg @devigned Thanks for the feedback. Closing this PR with respect to https://github.com/Azure/autorest/pull/1263 that'll avoid failures on unknown properties of the object while serializing.

**Pull request closed without merging**